File: 2766-relocate-marbles/2766-relocate-marbles.cpp

class Solution {
public:
    std::vector<int> relocateMarbles(std::vector<int>& nums, std::vector<int>& moveFrom, std::vector<int>& moveTo) {
        std::unordered_set<int> op;
        for (int el : nums) {
            op.insert(el);
        }
        for (int lp = 0; lp < moveFrom.size(); lp++) {
            if (op.count(moveFrom[lp])) {
                op.erase(moveFrom[lp]);
                op.insert(moveTo[lp]);
            }
        }
        std::vector<int> ans(op.begin(), op.end());
        std::sort(ans.begin(), ans.end());
        return ans;
    }
};

File: 2766-relocate-marbles/2766-relocate-marbles.java
class Solution {
       public static List<Integer> relocateMarbles(int[] arr, int[] moveFrom, int[] moveTo) {
Set<Integer> op = new HashSet<>();
        for (int el : arr) {
            op.add(el);
        }
        for (int lp = 0; lp < moveFrom.length; lp++) {
            if (op.contains(moveFrom[lp])) {
                op.remove(moveFrom[lp]);
                op.add(moveTo[lp]);
            }
        }
        List<Integer> ans = new ArrayList<>(op);
        Collections.sort(ans);
        return ans;
    }
}